amplipi/ctrl.py

```diff
@@ -651,6 +653,24 @@ def _get_source_config(self, sources: Optional[List[models.Source]] = None) -> L
       src_cfg[s] = self._is_digital(src.input)
     return src_cfg
 
+  def _delete_unused_temporary_streams(self):
+    """Removes temporary file players if they are disconnected and have no connected sources"""
+    temp_streams = []
+    for stream in self.status.streams:
+      if stream.type == 'fileplayer' and stream.temporary and stream.timeout is not None:
+        time = datetime.datetime.strptime(stream.timeout, '%Y-%m-%d %H:%M:%S.%f')
+        if time < datetime.datetime.now():
+          temp_streams.append(stream)
```

Comment:
I feel like this is better accessed via the specific stream class with a method like expired()